core/src/main/scala/za/co/absa/spline/harvester/builder/read/ReadCommandExtractor.scala

```diff
@@ -82,11 +82,26 @@ class ReadCommandExtractor(pathQualifier: PathQualifier, session: SparkSession)
         }
 
         case `_: CassandraSourceRelation`(casr) =>
-          val tableRef = extractFieldValue[AnyRef](casr,"tableRef")
+          val tableRef = extractFieldValue[AnyRef](casr, "tableRef")
           val table = extractFieldValue[AnyRef](tableRef, "table")
           val keyspace = extractFieldValue[AnyRef](tableRef, "keyspace")
           ReadCommand(SourceIdentifier.forCassandra(keyspace.asInstanceOf[String], table.asInstanceOf[String]), operation)
 
+        case `_: MongoDBSourceRelation`(mongr) =>
+          val mongoRDD = extractFieldValue[AnyRef](mongr, "mongoRDD")
+          val readConfig = extractFieldValue[AnyRef](mongoRDD, "readConfig")
+          val database = extractFieldValue[AnyRef](readConfig, "databaseName")
+          val collection = extractFieldValue[AnyRef](readConfig, "collectionName")
+          val connectionUrlx = extractFieldValue[AnyRef](readConfig, "connectionString")
+
+          val connectionUrl_Scala_2_11 = "x"
```

Comment:
This slipped out of my sight. Those `x` and `value`, aren't you are dealing with `Option` monad here?
Can we use more strict types instead of `AnyRef` that would be compatible between Spark versions?

Reply:
Please take a look at my commit here. Do you agree with that? 4bf769f1e78a2edca94dbd86cc7129c0267ef29b

Reply:
Yes got it. I'll try and implement future ones like this instead of the way I was doing it.